docs/docs/ref-02-component-api.md

```diff
@@ -14,10 +14,21 @@ Instances of a React Component are created internally in React when rendering. T
 ### setState
 
 ```javascript
-setState(object nextState[, function callback])
+setState(function|object nextState[, function callback])
 ```
 
-Merges nextState with the current state. This is the primary method you use to trigger UI updates from event handlers and server request callbacks.  In addition, you can supply an optional callback function that is executed once `setState` is completed and the component is re-rendered.
+Merges nextState with the current state. This is the primary method you use to trigger UI updates from event handlers and server request callbacks.
+
+The first argument can be an object (containing zero or more keys to update) or a function (of state and props) that returns an object containing keys to update.  The following are equivilent:
+
+```javascript
+setState({mykey: 'my new value'});
+setState(function(previousState, currentProps){return {mykey: 'my new value'};});
+```
+
+The former syntax is simpler and sufficient in most cases, while the later syntax is slightly more powerful (allowing you to enqueue an atomic update that consults the previous value of state+props before setting any values).
```

Comment:
Let's also reword this a little bit. I'd like the primary focus of this documentation to be on the simple object usage, since it is overwhelmingly what people will be doing. As is they sort of have equal footing.

> here's the simple object usage...
> ...
> It's also possible to pass a function with the signature (...). This can be useful in some cases because of reasons and here's an example demonstrating reasons